rfd/0229-agent-inventory-ui.md

```diff
@@ -0,0 +1,118 @@
+---
+authors: Yassine Bounekhla (yassine@goteleport.com)
+state: draft
+---
+
+# RFD 0229 - Agent Inventory UI
+
+## Required Approvals
+
+- Engineering: @zmb3 @fspmarshall
+- Product: @roraback
+
+## What
+
+A page in the Web UI that lists connected agents and their details, with options to search and filter through them.
+
+## Why
+
+Currently, the only way to list agents is via the `tctl inventory list` CLI command, this can be quite cumbersome for users 
+and as a result limits the visibility cluster administrators have into their infrastructure. Exposing this information in the 
+Web UI would greatly improve the user experience and make it easier for cluster administrators to do things like identify agents
+still running old versions of Teleport and need to be updated.
+
+## Details
+
+### Implementation
+
+The first version of this feature will primarily focus on adding the existing `tctl inventory` functionality to the Web UI and use
+most of the same underlying backend API.
+
+#### Web Endpoint
+
+A new `/webapi/sites/:site/inventory` will be used by the Web UI to fetch a paginated list of agents, this endpoint will support 
+various query parameters for pagination and filtering, including:
+
+- `search`: Filter by hostname.
+- `versions`: Filter to only return agents running the specified Teleport version(s).
```

Comment:
Is this just for a specific version?

`tctl inventory ls` has options for `--exact-version`, `--newer-than`, and `--older-than`, which are helpful for finding outdated agents.

Reply:
This would let you select one or more from the list of versions that are currently found on your argents. It seemed a little excessive to replicate the `tctl` filters exactly in the UI as that would mean 3 different controls in the UI just for version filtering. I'll sync with @roraback and see if we can come up with something in the middle.